From 4d62612c5a22d1126d71c5c9caf569c86cd268d5 Mon Sep 17 00:00:00 2001
From: Tatsuo ONISHI <onishi@latmos.ipsl.fr>
Date: Mon, 10 Oct 2022 12:37:34 +0200
Subject: [PATCH 35/71] Dry deposition on snow and ice

Deposition over snow and ice added in CBMZ-MOSAIC and SAPRC-MOSAIC mechanisms

Source : Louis Marelle
TYPE : Enhancement

list of modified files:
---------------------------------------------------------------------------------
chem/module_dep_simple.F
   * Deposition over snow and ice xice added in CBMZ-MOSAIC and SAPRC-MOSAIC mechanisms
     wrf_season external file
        set iseason = 1 or 2 also for mechanisms CBMZ-MOSAIC and SAPRC99-MOSAIC
        Later, for xice(i,j) >= 0.15, set iland = 8 and iseason = 4
        The snippet below is intended only for the northern hemisphere.

        if(julday < 90 .or. julday > 270) then
          iseason = 2
        else
          iseason = 1
        endif

chem/dry_dep_driver.F
   * Deposition over ice : xice is passed to wesely_driver in chem/module_dep_simple.F (already done before this commit)
---
 chem/module_dep_simple.F | 28 +++++++++++++++++++++++++++-
 1 file changed, 27 insertions(+), 1 deletion(-)

diff --git a/chem/module_dep_simple.F b/chem/module_dep_simple.F
index f11ab9b..5bfc9ec 100755
--- a/chem/module_dep_simple.F
+++ b/chem/module_dep_simple.F
@@ -224,7 +224,33 @@ tile_lon_loop : &
                   iseason = 4
                endif
             end if
-
+!lm Also use the wrf_season external file for setting iseason when using
+! CBMZ-MOSAIC and SAPRC99-MOSAIC mechanisms. 
+            if( config_flags%chem_opt == CBMZ_MOSAIC_4BIN .or. &
+                config_flags%chem_opt == CBMZ_MOSAIC_8BIN .or. &
+                config_flags%chem_opt == CBMZ_MOSAIC_4BIN_AQ .or. &
+                config_flags%chem_opt == CBMZ_MOSAIC_8BIN_AQ .or. &
+                config_flags%chem_opt == CBMZ_MOSAIC_DMS_4BIN .or. &
+                config_flags%chem_opt == CBMZ_MOSAIC_DMS_8BIN .or. &
+                config_flags%chem_opt == CBMZ_MOSAIC_KPP .or. &
+                config_flags%chem_opt == CBMZ_MOSAIC_DMS_4BIN_AQ .or. &
+                config_flags%chem_opt == CBMZ_MOSAIC_DMS_8BIN_AQ .or. &
+                config_flags%chem_opt == SAPRC99_MOSAIC_4BIN_VBS2_KPP .or. &
+                config_flags%chem_opt == SAPRC99_MOSAIC_8BIN_VBS2_AQ_KPP .or. &
+                config_flags%chem_opt == SAPRC99_MOSAIC_8BIN_VBS2_KPP) then
+               ! Add cases for deposition over snow and ice in CBMZ-MOSAIC and
+               ! SAPRC-MOSAIC mechanisms
+               if( snowh(i,j) < .01 ) then
+                  iseason = seasonal_pft(id)%seasonal_wes(i,j,iland,current_month)
+               else
+                 iseason = 4
+               endif
+               if( xice(i,j) >= .15 ) then
+                 iland = 8
+                 iseason = 4
+               endif
+            end if
+!lm
             ta    = tsk(i,j)
             rad   = gsw(i,j)
             vegfrac = vegfra(i,j)
-- 
2.31.1

